.github/workflows/main.yml

```diff
@@ -26,4 +26,6 @@ jobs:
         sudo apt-get install libsodium-dev
 
     - name: Test compilation
-      run: dub build
+      run: |
+        echo "AGORA_VERSION=HEAD" >> $GITHUB_ENV
+        dub build
```

Comment:
For some reasons, Github decided that versions exported this way will only be accessible from the *next* step.

Reply:
yes, that is in line with the docs:
https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable

how about:
```
AGORA_VERSION=HEAD dub build
```

Reply:
Yeah this works, and so does:
```yaml
- name: Test compilation
   env:
     - AGORA_VERSION: HEAD
   run: dub build
```